pkg/kube/client.go

```diff
@@ -628,3 +641,89 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
 
 	return err
 }
+
+//get the relation pods of kubernetes resources
+// kubernetes resource used select labels to relate pods
+func (c *Client) getRelationPods(infos []*resource.Info) map[string][]api.Pod {
+	var objPods = make(map[string][]api.Pod)
+
+	for _, value := range infos {
+		objPods, _ = c.getSelectRelationPod(value, objPods)
+	}
+	return objPods
+}
+
+//get an kubernetes resources's relation pods
+func (c *Client) getSelectRelationPod(info *resource.Info, objPods map[string][]api.Pod) (map[string][]api.Pod, error) {
+	if info == nil {
+		return objPods, nil
+	}
+
+	err := info.Get()
+	if err != nil {
+		return objPods, err
+	}
+
+	versioned, err := c.AsVersionedObject(info.Object)
+	if runtime.IsNotRegisteredError(err) {
+		return objPods, nil
+	}
+	if err != nil {
+		return objPods, err
+	}
+
+	selector, err := getSelectorFromObject(versioned)
+
+	c.Log("getSelectRelationPod selector: %+v", selector)
+	if err != nil {
+		return objPods, err
+	}
+	client, _ := c.ClientSet()
+
+	pods, err := client.Core().Pods(info.Namespace).List(metav1.ListOptions{
+		FieldSelector: fields.Everything().String(),
+		LabelSelector: labels.Set(selector).AsSelector().String(),
+	})
+	if err != nil {
+		return objPods, err
+	}
+
+	for _, pod := range pods.Items {
+		c.Log("get select relation pod: %s/%s", pod.Namespace, pod.Name)
```

Comment:
This feels more like debug information. I would remove this log and the one on line 677 and create a log statement at the top of function saying that you are fetching pod info for the object

Reply:
I got it, many thanks.  I will modify it in this weekend.